複数行一括変換のサポートなど

tokorom opened the pull request:
@laprasdrum Chrome Extensionの作成ありがとうございました！！

これから是非活用をさせてください。
また、とても感銘を受けて（今までChrome Extensionを作ったことがありませんでした）実運用に合わせて以下のカスタマイズをさせていただきました。

- 複数行を一括で変換する
    - ただ、これをサポートする場合、空白区切りでの判定のままだと厳しいため、Wikiのmarkdown形式（`|`区切り）のものを選択して利用させていただくことにしました

- presenterNameにはTwitterアカウントの記載があればそれを使い、なければアカウント名を使う
    - これまでもやりたかったけど手運用だと面倒でサボっていた部分でした

- Twitterアカウントがリンク付きの記述の場合はTwitterアカウントのみ抽出する
    - Markdown形式でコピーするようにした副作用の回避

tokorom committed:
Support multiline

laprasdrum commented:
> Wikiのmarkdown形式（|区切り）のものを選択して利用させていただくことにしました

お、なるほど
たしかにこっちのほうが parse しやすそうですね ✨ 

利用ケースとして README もいじったほうがいいかなと思いましたが、他の運営者にバトンタッチするときが来たときでもよさそうですね

laprasdrum commented on background.js:
```diff
@@ -12,42 +17,84 @@ function saveToClipboard(text) {
+    let twitterAccount = extractTwitterAccount(values[4].trim());
+    let docURL = values[5].trim();
+
+    let presenter = (twitterAccount.length == 0) ? accountName : twitterAccount;
```
テキスト長で分岐するよりは、「twitterAccount が存在しないなら accountName を使う」という表現のほうが直感的でよさそうですね。

例えば `extractTwitterAccount` の返り値を「一文字以上のテキスト、もしくは null」とすれば
```js
let presenter = twitterAccount || accountName;
```
と書けますね 👀

tokorom replied:
おお `||` でできるんですね！

laprasdrum commented on background.js:
```diff
@@ -12,42 +17,84 @@ function saveToClipboard(text) {
+    let docURL = values[5].trim();
+
+    let presenter = (twitterAccount.length == 0) ? accountName : twitterAccount;
+    let nicknameText = (nickname.length == 0) ? "" : "（" + nickname + "）";
```
こちらも `presenter` の件と同じですね

laprasdrum reviewed (approved):
動作は大丈夫そうです 👍 
コメントしていますが修正はおまかせします 🙏

tokorom committed:
After the code review

https://github.com/laprasdrum/potatotips-tweet-generator/pull/1#pullrequestreview-148791798

tokorom commented:
@laprasdrum レビューありがとうございました！ご指摘いただいた点を修正しました。

`"" || "foo"` で空文字を回避できるのが未知で勉強になりました。

laprasdrum commented:
LGTM ですー 🎉 
merge しちゃいますねー
